src/generic/eigen_solver.cc

```diff
@@ -700,19 +736,205 @@ namespace oomph
+  /// while \f$ \beta_i \f$ is real. The actual eigenvalues may then be
+  /// computed by doing the division, checking for zero betas to avoid NaNs.
+  /// There's a convenience wrapper to this function that simply computes
+  /// these eigenvalues regardless. That version may die in NaN checking is
+  /// enabled (via the fenv.h header and the associated feenable function).
+  /// At least n_eval eigenvalues are computed.
+  //==========================================================================
+  void LAPACK_QZ::solve_eigenproblem(
+    Problem* const& problem_pt,
+    const int& n_eval,
+    Vector<std::complex<double>>& alpha_eval,
+    Vector<double>& beta_eval,
+    Vector<Vector<std::complex<double>>>& eigenvector)
+  {
+    Vector<DoubleVector> eigenvector_aux;
+
+    // Call raw interface to lapack qz
+    solve_eigenproblem_helper(
+      problem_pt, n_eval, alpha_eval, beta_eval, eigenvector_aux);
+
+    // Now move auxiliary data into actual complex eigenvalues
+    // Instrutions from lapack qz (where VR translates into eigenvector_aux):
+    //        VR is DOUBLE PRECISION array, dimension (LDVR,N)
+    //        If JOBVR = 'V', the right eigenvectors v(j) are stored one
+    //        after another in the columns of VR, in the same order as
+    //        their eigenvalues. If the j-th eigenvalue is real, then
+    //        v(j) = VR(:,j), the j-th column of VR. If the j-th and
+    //        (j+1)-th eigenvalues form a complex conjugate pair, then
+    //        v(j) = VR(:,j)+i*VR(:,j+1) and v(j+1) = VR(:,j)-i*VR(:,j+1).
+    //        Each eigenvector is scaled so the largest component has
+    //        abs(real part)+abs(imag. part)=1.
+    unsigned n = problem_pt->ndof();
+    eigenvector.resize(n);
+    unsigned eval_count = 0;
+    while (eval_count < n)
+    {
+      // i-th eigenvalue is real:
+      if (alpha_eval[eval_count].imag() == 0.0)
+      {
+        // Resize the single eigenvector associated with this
+        // single real eigenvalue
+        eigenvector[eval_count].resize(n);
+        for (unsigned j = 0; j < n; ++j)
+        {
+          eigenvector[eval_count][j] =
+            std::complex<double>(eigenvector_aux[j][eval_count], 0.0);
+        }
+        eval_count++;
+      }
+      // Assume (and check!) that complex conjugate pairs follow each other
+      // as implied by
+      // http://www.netlib.org/lapack/explore-html/d9/d8e/group__double_g_eeigen_ga4f59e87e670a755b41cbdd7e97f36bea.html
+      else
+      {
+#ifdef PARANOID
+
+        // Are the eigenvalues finite?
+        if ((beta_eval[eval_count] != 0.0) &&
+            (beta_eval[eval_count + 1] != 0.0))
+        {
+          std::complex<double> lambda_this =
+            alpha_eval[eval_count] / beta_eval[eval_count];
+
+          std::complex<double> lambda_next =
+            alpha_eval[eval_count + 1] / beta_eval[eval_count + 1];
+
+          // Check cc-ness to within tolerance
```

Comment:
Modify comment to indicate that this is a check for failure of cc-ness?

Reply:
Done.